src/main/java/com/redhat/devtools/intellij/tektoncd/actions/OpenEditorAction.java

```diff
@@ -16,87 +16,44 @@
+    }
 
 
     @Override
     public void actionPerformed(AnActionEvent anActionEvent, TreePath path, Object selected, Tkn tkncli) {
         ParentableNode<? extends ParentableNode<NamespaceNode>> element = getElement(selected);
         String content = "";
         String namespace = element.getParent().getParent().getName();
-        String kind = "";
+        Pair<String, String> yamlAndKind = null;
         try {
-            if (element instanceof PipelineNode) {
-                content = tkncli.getPipelineYAML(namespace, element.getName());
-                kind = KIND_PIPELINES;
-            } else if (element instanceof ResourceNode) {
-                content = tkncli.getResourceYAML(namespace, element.getName());
-                kind = KIND_RESOURCES;
-            } else if (element instanceof TaskNode) {
-                content = tkncli.getTaskYAML(namespace, element.getName());
-                kind = KIND_TASKS;
-            } else if (element instanceof ClusterTaskNode) {
-                content = tkncli.getClusterTaskYAML(element.getName());
-                kind = KIND_CLUSTERTASKS;
-            } else if (element instanceof ConditionNode) {
-                content = tkncli.getConditionYAML(namespace, element.getName());
-                kind = KIND_CONDITIONS;
-            } else if (element instanceof TriggerTemplateNode) {
-                content = tkncli.getTriggerTemplateYAML(namespace, element.getName());
-                kind = KIND_TRIGGERTEMPLATES;
-            } else if (element instanceof TriggerBindingNode) {
-                content = tkncli.getTriggerBindingYAML(namespace, element.getName());
-                kind = KIND_TRIGGERBINDINGS;
-            } else if (element instanceof ClusterTriggerBindingNode) {
-                content = tkncli.getClusterTriggerBindingYAML(namespace, element.getName());
-                kind = KIND_CLUSTERTRIGGERBINDINGS;
-            } else if (element instanceof EventListenerNode) {
-                content = tkncli.getEventListenerYAML(namespace, element.getName());
-                kind = KIND_EVENTLISTENER;
-            }
-        }
-        catch (IOException e) {
+            yamlAndKind = TreeHelper.getYAMLAndKindFromNode(element);
+        } catch (IOException e) {
             UIHelper.executeInUI(() -> Messages.showErrorDialog("Error: " + e.getLocalizedMessage(), "Error"));
         }
 
-        if (!content.isEmpty()) {
+        if (yamlAndKind != null && !yamlAndKind.first.isEmpty()) {
             Project project = anActionEvent.getProject();
 
             Optional<FileEditor> editor = Arrays.stream(FileEditorManager.getInstance(project).getAllEditors()).
-                    filter(fileEditor -> fileEditor.getFile().getName().startsWith(namespace + "-" + element.getName()) &&
+                    filter(fileEditor -> fileEditor.getFile().getName().startsWith(namespace + "-" + element.getName() + ".yaml") &&
```

Comment:
I guess this can be edited in equals() and `fileEditor.getFile().getExtension().equals("yaml")` can be removed if we explicitly declare the extension in the first check. I guess we didn't add the ".yaml" because we were using another way to display resources and a number got appended (e.g "res_1.yaml, res_2.yaml") but we switched to temp file.

Reply:
OK, I change that, because I face problem when I have two pipeline which starts with same name, I wasn't able to open second(`foo.yam`), with longer name if first was open(`fooBar.yam`).